mpqp/execution/result.py

```diff
@@ -258,6 +258,9 @@ def __init__(
                 )
             else:
                 self._state_vector = data
+                if job.circuit.gphase != 0:
```

Comment:
is this check really needed ?

Reply:
In case there is no custom gate (which I assume will be the most common usage), I think checking that the phase is null is less costful than calling `np.exp` independently of the phase.